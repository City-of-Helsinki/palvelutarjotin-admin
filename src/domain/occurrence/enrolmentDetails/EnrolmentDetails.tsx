import classNames from 'classnames';
import { format } from 'date-fns';
import {
  Button,
  IconArrowLeft,
  IconCheck,
  IconCross,
  IconCrossCircle,
  IconPen,
} from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router';
import { toast } from 'react-toastify';

import LoadingSpinner from '../../../common/components/loadingSpinner/LoadingSpinner';
import {
  EnrolmentStatus,
  OccurrenceDocument,
  OccurrenceQuery,
  useApproveEnrolmentMutation,
  useDeclineEnrolmentMutation,
  useDeleteEnrolmentMutation,
  useEnrolmentQuery,
} from '../../../generated/graphql';
import useLocale from '../../../hooks/useLocale';
import { translateValue } from '../../../utils/translateUtils';
import { ROUTES } from '../../app/routes/constants';
import ApproveEnrolmentModal from '../enrolmentTable/enrolmentModals/ApproveEnrolmentModal';
import DeclineEnrolmentModal from '../enrolmentTable/enrolmentModals/DeclineEnrolmentModal';
import DeleteEnrolmentModal from '../enrolmentTable/enrolmentModals/DeleteEnrolmentModal';
import styles from './enrolmentDetails.module.scss';
import EnrolmentInfoRow from './EnrolmentInfoRow';
import { getNotificationInfoText } from './utils';

const EnrolmentDetails: React.FC<{
  enrolmentId: string;
  occurrenceId: string;
  eventId: string;
  onGoBackClick: () => void;
}> = ({ enrolmentId, occurrenceId, eventId, onGoBackClick }) => {
  const history = useHistory();
  const locale = useLocale();
  const { t } = useTranslation();
  const [approveModalOpen, setApproveModalOpen] = React.useState(false);
  const [declineModalOpen, setDeclineModalOpen] = React.useState(false);
  const [deleteModalOpen, setDeleteModalOpen] = React.useState(false);
  const { data: enrolmentData, loading: loadingEnrolment } = useEnrolmentQuery({
    variables: {
      id: enrolmentId,
    },
  });
  const enrolment = enrolmentData?.enrolment;
  const enrolmentIsNotApproved = enrolment?.status !== EnrolmentStatus.Approved;
  const enrolmentIsNotDeclined = enrolment?.status !== EnrolmentStatus.Declined;

  const [
    approveEnrolment,
    { loading: loadingApproveEnrolment },
  ] = useApproveEnrolmentMutation({
    onError: (error) => {
      toast(t('enrolment.approveEnrolmentError'), {
        type: toast.TYPE.ERROR,
      });
    },
    onCompleted: () => setApproveModalOpen(false),
  });

  const [
    declineEnrolment,
    { loading: loadingDeclineEnrolment },
  ] = useDeclineEnrolmentMutation({
    onError: (error) => {
      toast(t('enrolment.declineEnrolmentError'), {
        type: toast.TYPE.ERROR,
      });
    },
    onCompleted: () => setDeclineModalOpen(false),
  });

  const [deleteEnrolment] = useDeleteEnrolmentMutation({
    onError: (error) => {
      toast(t('enrolment.deleteEnrolmentError'), {
        type: toast.TYPE.ERROR,
      });
    },
    onCompleted: () => {
      setDeleteModalOpen(false);
      onGoBackClick();
    },
  });

  const handleApproveEnrolment = async () => {
    approveEnrolment({ variables: { input: { enrolmentId } } });
  };

  const handleDeclineEnrolment = () => {
    declineEnrolment({ variables: { input: { enrolmentId } } });
  };

  const handleDeleteEnrolment = async () => {
    if (enrolment) {
      await deleteEnrolment({
        variables: {
          input: { occurrenceId, studyGroupId: enrolment.studyGroup.id },
        },
        update: (cache) => {
          const occurrenceData = cache.readQuery<OccurrenceQuery>({
            query: OccurrenceDocument,
            variables: { id: occurrenceId },
          });
          const occurrence = occurrenceData?.occurrence;
          // overwrite occurrence from cache (delete enrolment)
          cache.writeQuery({
            query: OccurrenceDocument,
            data: {
              occurrence: {
                ...occurrence,
                enrolments: {
                  ...occurrence?.enrolments,
                  edges: occurrence?.enrolments.edges.filter(
                    (e) => e?.node?.id !== enrolment.id
                  ),
                },
              },
            },
          });
        },
      });
    }
  };

  const getStudyLevel = () => {
    const studyLevel = enrolment?.studyGroup?.studyLevel;
    if (studyLevel) {
      return studyLevel.startsWith('GRADE')
        ? t('studyLevel.grade_interval', {
            postProcess: 'interval',
            count: Number(studyLevel.split('_')[1]),
          })
        : translateValue('studyLevel.', studyLevel, t);
    }
  };

  const handleEditEnrolment = () => {
    history.push(
      `/${locale}${ROUTES.EDIT_ENROLMENT}`
        .replace(':eventId', eventId)
        .replace(':enrolmentId', enrolmentId)
    );
  };

  return (
    <LoadingSpinner isLoading={loadingEnrolment}>
      {enrolment ? (
        <div className={styles.enrolmentDetails}>
          <div className={styles.backButtonContainer}>
            <button
              aria-label={t(
                'enrolment.enrolmentDetails.buttonGoToEnrolleesList'
              )}
              onClick={onGoBackClick}
            >
              <IconArrowLeft />
            </button>
          </div>

          <div className={styles.detailsSection}>
            <div className={styles.topRow}>
              <div>
                <span className={styles.enrolleesTitle}>
                  {t('enrolment.enrolmentDetails.buttonEnrolleesTitle')}
                </span>
              </div>
              <ApproveEnrolmentModal
                isOpen={approveModalOpen}
                onClose={() => setApproveModalOpen(false)}
                // TODO: Will there be a way to approve many at the same time?
                enrollees={enrolment.person ? [enrolment?.person] : undefined}
                approveEnrolment={handleApproveEnrolment}
                loading={loadingApproveEnrolment}
              />
              <DeclineEnrolmentModal
                isOpen={declineModalOpen}
                onClose={() => setDeclineModalOpen(false)}
                // TODO: Will there be a way to decline many at the same time?
                enrollees={enrolment.person ? [enrolment?.person] : undefined}
                declineEnrolment={handleDeclineEnrolment}
                loading={loadingDeclineEnrolment}
              />
              <DeleteEnrolmentModal
                isOpen={deleteModalOpen}
                onClose={() => setDeleteModalOpen(false)}
                deleteEnrolment={handleDeleteEnrolment}
              />
              <div className={styles.actionButtons}>
                {enrolmentIsNotApproved && (
                  <Button
                    className={classNames(
                      styles.approveButton,
                      styles.actionButton
                    )}
                    onClick={() => setApproveModalOpen(true)}
                    variant="secondary"
                    iconLeft={<IconCheck />}
                    disabled={loadingApproveEnrolment}
                  >
                    {t('enrolment.enrolmentDetails.buttonApproveEnrolment')}
                  </Button>
                )}
                {enrolmentIsNotDeclined && (
                  <Button
                    className={classNames(
                      styles.declineButton,
                      styles.actionButton
                    )}
                    onClick={() => setDeclineModalOpen(true)}
                    variant="secondary"
                    iconLeft={<IconCross />}
                    disabled={loadingDeclineEnrolment}
                  >
                    {t('enrolment.enrolmentDetails.buttonDeclineEnrolment')}
                  </Button>
                )}
                <Button
                  className={styles.editButton}
                  onClick={handleEditEnrolment}
                  variant="secondary"
                  iconLeft={<IconPen />}
                >
                  {t('enrolment.enrolmentDetails.buttonEditEnrolment')}
                </Button>
                <Button
                  className={classNames(
                    styles.deleteButton,
                    styles.actionButton
                  )}
                  onClick={() => setDeleteModalOpen(true)}
                  variant="secondary"
                  iconLeft={<IconCrossCircle />}
                >
                  {t('enrolment.enrolmentDetails.buttonDeleteEnrolment')}
                </Button>
              </div>
            </div>
            <table className={styles.detailsTable}>
              <tbody>
                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelEnrolled')}
                  value={format(
                    new Date(enrolment?.enrolmentTime),
                    'dd.MM.yyyy hh:mm'
                  )}
                />
                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelStatus')}
                  value={translateValue(
                    'enrolment.status.',
                    enrolment.status,
                    t
                  )}
                />

                {enrolment.person && (
                  <>
                    <EnrolmentInfoRow
                      label={t('enrolment.enrolmentDetails.labelName')}
                      value={enrolment.person.name}
                      space
                    />
                    <EnrolmentInfoRow
                      label={t('enrolment.enrolmentDetails.labelEmail')}
                      value={enrolment.person.emailAddress}
                    />
                    <EnrolmentInfoRow
                      label={t('enrolment.enrolmentDetails.labelPhoneNumber')}
                      value={enrolment.person.phoneNumber}
                    />
                  </>
                )}

                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelStudyGroupName')}
                  value={enrolment.studyGroup.name}
                  space
                />
                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelGroupName')}
                  value={enrolment.studyGroup.groupName}
                />
                {enrolment.studyGroup.studyLevel && (
                  <EnrolmentInfoRow
                    label={t('enrolment.enrolmentDetails.labelStudyLevel')}
                    value={getStudyLevel()}
                  />
                )}

                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelResponsiblePerson')}
                  value={enrolment.studyGroup.person.name}
                  space
                />
                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelEmail')}
                  value={enrolment.studyGroup.person.emailAddress}
                />
                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelPhoneNumber')}
                  value={enrolment.studyGroup.person.phoneNumber}
                />

                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelNotifications')}
                  value={getNotificationInfoText(enrolment, t)}
                  space
                />
                <EnrolmentInfoRow
                  label={t('enrolment.enrolmentDetails.labelAdditionalInfo')}
                  value={enrolment.studyGroup.extraNeeds}
                  space
                />
              </tbody>
            </table>
          </div>
        </div>
      ) : (
        <div>
          <button
            aria-label={t('enrolment.enrolmentDetails.buttonGoToEnrolleesList')}
            onClick={onGoBackClick}
          >
            <IconArrowLeft />
          </button>
          <p>{t('enrolment.enrolmentDetails.error')}</p>
        </div>
      )}
    </LoadingSpinner>
  );
};

export default EnrolmentDetails;
