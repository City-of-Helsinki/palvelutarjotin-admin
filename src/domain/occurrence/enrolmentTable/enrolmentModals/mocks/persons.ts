const persons = [
  {
    personName: 'Nimi Niminen',
  },
  {
    personName: 'Toka Tokanen',
  },
  {
    personName: 'Kolme Kolmonen',
  },
];

export default persons;
