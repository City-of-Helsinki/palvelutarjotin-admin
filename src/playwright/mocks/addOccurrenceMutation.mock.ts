import {
  type AddOccurrenceMutation,
  OccurrencesOccurrenceSeatTypeChoices,
} from '../../generated/graphql';

export const MOCK_ADD_OCCURRENCE_MUTATION_RESPONSE: AddOccurrenceMutation = {
  addOccurrence: {
    occurrence: {
      id: 'T2NjdXJyZW5jZU5vZGU6MzIzNQ==',
      pEvent: {
        id: 'UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjE1NjY=',
        __typename: 'PalvelutarjotinEventNode',
      },
      amountOfSeats: 80,
      minGroupSize: 2,
      maxGroupSize: 10,
      seatsTaken: 0,
      seatsApproved: 0,
      seatType: OccurrencesOccurrenceSeatTypeChoices.ChildrenCount,
      remainingSeats: 80,
      languages: {
        edges: [
          {
            node: {
              id: 'en',
              name: 'English',
              __typename: 'LanguageNode',
            },
            __typename: 'LanguageNodeEdge',
          },
          {
            node: {
              id: 'fi',
              name: 'Finnish',
              __typename: 'LanguageNode',
            },
            __typename: 'LanguageNodeEdge',
          },
          {
            node: {
              id: 'sv',
              name: 'Swedish',
              __typename: 'LanguageNode',
            },
            __typename: 'LanguageNodeEdge',
          },
        ],
        __typename: 'LanguageNodeConnection',
      },
      startTime: '2200-06-21T05:30:00+00:00',
      endTime: '2200-06-21T06:00:00+00:00',
      placeId: 'tprek:7254',
      cancelled: false,
      __typename: 'OccurrenceNode',
    },
    __typename: 'AddOccurrenceMutationPayload',
  },
};
