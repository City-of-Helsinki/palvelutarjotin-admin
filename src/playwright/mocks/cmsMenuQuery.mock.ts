import type { MenuQuery } from '../../generated/graphql-cms';

export const MOCK_EMPTY_CMS_MENU_QUERY_RESPONSE: MenuQuery = {
  __typename: 'RootQuery',
  menu: {
    __typename: 'Menu',
    id: 'empty-menu-id',
    name: 'Empty Menu',
    slug: 'empty-menu-slug',
    menuId: 1,
    menuItems: {
      __typename: 'MenuToMenuItemConnection',
      nodes: [],
    },
  },
};
