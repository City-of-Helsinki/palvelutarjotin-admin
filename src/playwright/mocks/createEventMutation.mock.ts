import type { CreateEventMutation } from '../../generated/graphql';

export const MOCK_CREATE_EVENT_MUTATION_RESPONSE_ON_PAGE_1_SAVE: CreateEventMutation =
  {
    addEventMutation: {
      response: {
        statusCode: 201,
        body: {
          id: 'kultus:agls75a3jm',
          internalId:
            'https://linkedevents.api.test.hel.ninja/v1/event/kultus:agls75a3jm/',
          name: {
            en: '',
            fi: 'Testitapahtuman nimi',
            sv: '',
            __typename: 'LocalisedObject',
          },
          shortDescription: {
            en: '',
            fi: 'Lyhyt kuvaus',
            sv: '',
            __typename: 'LocalisedObject',
          },
          description: {
            en: '',
            fi: '<p>Pitempi kuvaus</p>\n',
            sv: '',
            __typename: 'LocalisedObject',
          },
          images: [],
          offers: [
            {
              isFree: false,
              description: {
                en: '',
                fi: 'Vain korttimaksu',
                sv: '',
                __typename: 'LocalisedObject',
              },
              price: {
                en: '10',
                fi: '10',
                sv: '10',
                __typename: 'LocalisedObject',
              },
              infoUrl: null,
              __typename: 'Offer',
            },
          ],
          pEvent: {
            id: 'UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjE1NjY=',
            neededOccurrences: 1,
            autoAcceptance: false,
            autoAcceptanceMessage: null,
            __typename: 'PalvelutarjotinEventNode',
          },
          infoUrl: {
            en: '',
            fi: 'https://example.org/testi/',
            sv: '',
            __typename: 'LocalisedObject',
          },
          __typename: 'Event',
        },
        __typename: 'EventMutationResponse',
      },
      __typename: 'AddEventMutation',
    },
  };
