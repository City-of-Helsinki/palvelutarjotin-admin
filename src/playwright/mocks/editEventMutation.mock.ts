import {
  type EditEventMutation,
  Language,
  OccurrencesOccurrenceSeatTypeChoices,
  OrganisationsOrganisationTypeChoices,
} from '../../generated/graphql';

// EditEvent mutation response on event creation page 2 to page 3 transition
// i.e. when the user clicks "Go to publishing" button on page 2:
export const MOCK_EDIT_EVENT_MUTATION_RESPONSE: EditEventMutation = {
  updateEventMutation: {
    response: {
      statusCode: 200,
      body: {
        id: 'kultus:agls75a3jm',
        internalId:
          'https://linkedevents.api.test.hel.ninja/v1/event/kultus:agls75a3jm/',
        name: {
          en: '',
          fi: 'Testitapahtuman nimi',
          sv: '',
          __typename: 'LocalisedObject',
        },
        shortDescription: {
          en: '',
          fi: 'Lyhyt kuvaus',
          sv: '',
          __typename: 'LocalisedObject',
        },
        description: {
          en: '',
          fi: '<p>Pitempi kuvaus</p>\n',
          sv: '',
          __typename: 'LocalisedObject',
        },
        images: [],
        infoUrl: {
          en: '',
          fi: 'https://example.org/testi/',
          sv: '',
          __typename: 'LocalisedObject',
        },
        pEvent: {
          id: 'UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjE1NjY=',
          nextOccurrenceDatetime: '2200-06-21T05:00:00+00:00',
          autoAcceptance: true,
          autoAcceptanceMessage: 'Peruutukset 2pv ennen',
          contactPerson: {
            id: 'UGVyc29uTm9kZTpjZjNkN2NlYS0xNGQxLTQzNmQtOTNmZS0zOTI4Mzc2ZTQ2YTE=',
            emailAddress: 'admin@example.com',
            name: 'Test-admin',
            phoneNumber: '',
            language: Language.Fi,
            placeIds: [],
            __typename: 'PersonNode',
          },
          contactEmail: 'admin@example.com',
          contactPhoneNumber: '+358-50-123456789',
          enrolmentEndDays: 2,
          enrolmentStart: '2199-12-31T14:30:00+00:00',
          externalEnrolmentUrl: null,
          isQueueingAllowed: true,
          neededOccurrences: 1,
          mandatoryAdditionalInformation: false,
          organisation: {
            id: 'T3JnYW5pc2F0aW9uTm9kZTox',
            name: 'Kulttuurin ja vapaa-ajan toimiala',
            persons: {
              edges: [
                {
                  node: {
                    id: 'UGVyc29uTm9kZTpjZjNkN2NlYS0xNGQxLTQzNmQtOTNmZS0zOTI4Mzc2ZTQ2YTE=',
                    emailAddress: 'admin@example.com',
                    name: 'Test-admin',
                    phoneNumber: '',
                    language: Language.Fi,
                    placeIds: [],
                    __typename: 'PersonNode',
                  },
                  __typename: 'PersonNodeEdge',
                },
              ],
              __typename: 'PersonNodeConnection',
            },
            phoneNumber: '',
            publisherId: 'ahjo:u480400',
            type: OrganisationsOrganisationTypeChoices.Provider,
            __typename: 'OrganisationNode',
          },
          occurrences: {
            edges: [
              {
                node: {
                  id: 'T2NjdXJyZW5jZU5vZGU6MzIzNQ==',
                  pEvent: {
                    id: 'UGFsdmVsdXRhcmpvdGluRXZlbnROb2RlOjE1NjY=',
                    __typename: 'PalvelutarjotinEventNode',
                  },
                  amountOfSeats: 80,
                  minGroupSize: 2,
                  maxGroupSize: 10,
                  seatsTaken: 0,
                  seatsApproved: 0,
                  seatType: OccurrencesOccurrenceSeatTypeChoices.ChildrenCount,
                  remainingSeats: 80,
                  languages: {
                    edges: [
                      {
                        node: {
                          id: 'en',
                          name: 'English',
                          __typename: 'LanguageNode',
                        },
                        __typename: 'LanguageNodeEdge',
                      },
                      {
                        node: {
                          id: 'fi',
                          name: 'Finnish',
                          __typename: 'LanguageNode',
                        },
                        __typename: 'LanguageNodeEdge',
                      },
                      {
                        node: {
                          id: 'sv',
                          name: 'Swedish',
                          __typename: 'LanguageNode',
                        },
                        __typename: 'LanguageNodeEdge',
                      },
                    ],
                    __typename: 'LanguageNodeConnection',
                  },
                  startTime: '2200-06-21T05:00:00+00:00',
                  endTime: '2200-06-21T06:00:00+00:00',
                  placeId: 'tprek:7254',
                  cancelled: false,
                  __typename: 'OccurrenceNode',
                },
                __typename: 'OccurrenceNodeEdge',
              },
            ],
            __typename: 'OccurrenceNodeConnection',
          },
          translations: [
            {
              autoAcceptanceMessage: 'Peruutukset 2pv ennen',
              languageCode: Language.Fi,
              __typename: 'PalvelutarjotinEventTranslationType',
            },
          ],
          __typename: 'PalvelutarjotinEventNode',
        },
        inLanguage: [],
        audience: [
          {
            id: null,
            name: null,
            internalId:
              'https://linkedevents.api.test.hel.ninja/v1/keyword/kultus:40/',
            __typename: 'Keyword',
          },
          {
            id: null,
            name: null,
            internalId:
              'https://linkedevents.api.test.hel.ninja/v1/keyword/kultus:58/',
            __typename: 'Keyword',
          },
        ],
        keywords: [
          {
            id: null,
            name: null,
            internalId:
              'https://linkedevents.api.test.hel.ninja/v1/keyword/kultus:17/',
            __typename: 'Keyword',
          },
          {
            id: null,
            name: null,
            internalId:
              'https://linkedevents.api.test.hel.ninja/v1/keyword/kultus:5/',
            __typename: 'Keyword',
          },
          {
            id: null,
            name: null,
            internalId:
              'https://linkedevents.api.test.hel.ninja/v1/keyword/yso:p18434/',
            __typename: 'Keyword',
          },
        ],
        startTime: '2025-06-02T09:23:35.687000Z',
        publicationStatus: 'draft',
        datePublished: null,
        endTime: null,
        offers: [
          {
            isFree: false,
            description: {
              en: '',
              fi: 'Vain korttimaksu',
              sv: '',
              __typename: 'LocalisedObject',
            },
            price: {
              en: '10',
              fi: '10',
              sv: '10',
              __typename: 'LocalisedObject',
            },
            infoUrl: null,
            __typename: 'Offer',
          },
        ],
        __typename: 'Event',
      },
      __typename: 'EventMutationResponse',
    },
    __typename: 'UpdateEventMutation',
  },
};
