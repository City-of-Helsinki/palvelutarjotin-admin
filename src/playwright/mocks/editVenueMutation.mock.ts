import { type EditVenueMutation, Language } from '../../generated/graphql';

// EditVenue mutation response on event creation page 2 to page 3 transition
// i.e. when the user clicks "Go to publishing" button on page 2:
export const MOCK_EDIT_VENUE_MUTATION_RESPONSE: EditVenueMutation = {
  updateVenue: {
    venue: {
      id: 'tprek:7254',
      hasClothingStorage: true,
      hasSnackEatingPlace: false,
      outdoorActivity: false,
      hasToiletNearby: true,
      hasAreaForGroupWork: true,
      hasIndoorPlayingArea: true,
      hasOutdoorPlayingArea: false,
      translations: [
        {
          languageCode: Language.Fi,
          description: 'Annantalo on taidekeskus Helsingissä.',
          __typename: 'VenueTranslationType',
        },
        {
          languageCode: Language.Sv,
          description: 'Annegården är ett konstcentrum i Helsingfors.',
          __typename: 'VenueTranslationType',
        },
        {
          languageCode: Language.En,
          description: 'Annantalo is an arts centre in Helsinki.',
          __typename: 'VenueTranslationType',
        },
      ],
      __typename: 'VenueNode',
    },
    __typename: 'UpdateVenueMutationPayload',
  },
};
