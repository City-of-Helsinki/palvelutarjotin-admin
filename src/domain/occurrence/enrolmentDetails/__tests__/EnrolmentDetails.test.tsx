import { MockedProvider } from '@apollo/react-testing';
import { render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import Router from 'react-router';

import { EnrolmentDocument } from '../../../../generated/graphql';
import EnrolmentDetails from '../EnrolmentDetails';
import { enrolmentResult } from './mocks';

const mocks = [
  {
    request: {
      query: EnrolmentDocument,
      variables: {
        id: 'RW5yb2xtZW50Tm9kZTo1Ng==',
      },
    },
    result: enrolmentResult,
  },
];

beforeEach(() => {
  jest.spyOn(Router, 'useParams').mockReturnValue({});
  jest
    .spyOn(Router, 'useLocation')
    .mockReturnValue({ pathname: '/', search: '', state: '', hash: '' });
  jest.spyOn(Router, 'useHistory').mockReturnValue({});
});

test('matches snapshot', async () => {
  const { container } = render(
    <MockedProvider mocks={mocks}>
      <EnrolmentDetails
        enrolmentId="RW5yb2xtZW50Tm9kZTo1Ng=="
        eventId=""
        occurrenceId=""
        onGoBackClick={jest.fn()}
      />
    </MockedProvider>
  );

  await waitFor(() => {
    expect(screen.queryByText('Ilmoittautuneet')).toBeInTheDocument();
  });

  expect(container).toMatchSnapshot();
});

test('renders correct information', async () => {
  render(
    <MockedProvider mocks={mocks}>
      <EnrolmentDetails
        enrolmentId="RW5yb2xtZW50Tm9kZTo1Ng=="
        eventId=""
        occurrenceId=""
        onGoBackClick={jest.fn()}
      />
    </MockedProvider>
  );

  await waitFor(() => {
    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
  });

  // values
  expect(screen.queryByText('14.08.2020 10:15')).toBeInTheDocument();
  expect(screen.queryByText('Hyväksytty')).toBeInTheDocument();
  // expect(screen.queryByText('Ilmoittautuja')).toBeInTheDocument();
  // expect(screen.queryByText('ilmo@ilmoittautuja.com')).toBeInTheDocument();
  expect(
    screen.queryByText('Tekstiviestillä, sähköpostilla, kieli: suomi')
  ).toBeInTheDocument();
  // expect(screen.queryByText('123321123')).toBeInTheDocument();
  expect(screen.queryByText('Yläaste')).toBeInTheDocument();
  expect(screen.queryByText('Ryhmän nimi')).toBeInTheDocument();
  // expect(screen.queryByText('ilmo@ilmoittautuja.com')).toBeInTheDocument();
  expect(screen.queryByText('Lisätietoja tässä')).toBeInTheDocument();
});
