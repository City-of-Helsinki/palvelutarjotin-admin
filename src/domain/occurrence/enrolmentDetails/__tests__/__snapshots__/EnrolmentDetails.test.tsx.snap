// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="enrolmentDetails"
  >
    <div
      class="backButtonContainer"
    >
      <button
        aria-label="Palaa ilmoittautuneisiin"
      >
        <svg
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <rect
              height="24"
              width="24"
            />
            <polygon
              fill="currentColor"
              points="10 5.5 11.5 7 7.5 11 20 11 20 13 7.5 13 11.5 17 10 18.5 3.5 12"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="detailsSection"
    >
      <div
        class="topRow"
      >
        <div>
          <span
            class="enrolleesTitle"
          >
            Ilmoittautuneet
          </span>
        </div>
        <div
          class="actionButtons"
        >
          <button
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS declineButton actionButton"
            type="button"
          >
            <div
              aria-hidden="true"
              class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
            >
              <svg
                class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <rect
                    height="24"
                    width="24"
                  />
                  <polygon
                    fill="currentColor"
                    points="18 7.5 13.5 12 18 16.5 16.5 18 12 13.5 7.5 18 6 16.5 10.5 12 6 7.5 7.5 6 12 10.5 16.5 6"
                  />
                </g>
              </svg>
            </div>
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Jätä ilman paikkaa
            </span>
          </button>
          <button
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS editButton"
            type="button"
          >
            <div
              aria-hidden="true"
              class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
            >
              <svg
                class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M0 0h24v24H0z"
                  />
                  <path
                    d="M16.75 3L4.5 15.25l.014 4.736L9.25 20 21.5 7.75 16.75 3zM6.5 16L16.75 5.75l2 2L8.5 18l-1.993-.006L6.5 16z"
                    fill="currentColor"
                  />
                </g>
              </svg>
            </div>
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Muokkaa
            </span>
          </button>
          <button
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS deleteButton actionButton"
            type="button"
          >
            <div
              aria-hidden="true"
              class="Button-module_icon__O-h7R button_hds-icon__17j8Z"
            >
              <svg
                class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <rect
                    height="24"
                    width="24"
                  />
                  <path
                    d="M12,2 C17.5228475,2 22,6.4771525 22,12 C22,17.5228475 17.5228475,22 12,22 C6.4771525,22 2,17.5228475 2,12 C2,6.4771525 6.4771525,2 12,2 Z M12,4 C7.581722,4 4,7.581722 4,12 C4,16.418278 7.581722,20 12,20 C16.418278,20 20,16.418278 20,12 C20,7.581722 16.418278,4 12,4 Z M15,7.5 L16.5,9 L13.5,12 L16.5,15 L15,16.5 L12,13.5 L9,16.5 L7.5,15 L10.5,12 L7.5,9 L9,7.5 L12,10.5 L15,7.5 Z"
                    fill="currentColor"
                  />
                </g>
              </svg>
            </div>
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Poista
            </span>
          </button>
        </div>
      </div>
      <table
        class="detailsTable"
      >
        <tbody>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Ilmoittautunut:
            </td>
            <td>
              14.08.2020 10:15
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Tila:
            </td>
            <td>
              Hyväksytty
            </td>
          </tr>
          <tr
            class="infoRow spacingRow"
          >
            <td
              class="infoLabel"
            >
              Nimi:
            </td>
            <td>
              Ilmoittautuja
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Sähköpostiosoite:
            </td>
            <td>
              ilmo@ilmoittautuja.com
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Puhelinnumero:
            </td>
            <td>
              123321123
            </td>
          </tr>
          <tr
            class="infoRow spacingRow"
          >
            <td
              class="infoLabel"
            >
              Päiväkoti / koulu / oppilaitos:
            </td>
            <td>
              Yläaste
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Ryhmän nimi:
            </td>
            <td>
              Ryhmän nimi
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Luokka-aste:
            </td>
            <td>
              5. luokka
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Lisätiedot:
            </td>
            <td>
              Lisätietoja tässä
            </td>
          </tr>
          <tr
            class="infoRow spacingRow"
          >
            <td
              class="infoLabel"
            >
              Vastuussa oleva aikuinen:
            </td>
            <td>
              Ilmoittautuja
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Sähköpostiosoite:
            </td>
            <td>
              ilmo@ilmoittautuja.com
            </td>
          </tr>
          <tr
            class="infoRow"
          >
            <td
              class="infoLabel"
            >
              Puhelinnumero:
            </td>
            <td>
              123321123
            </td>
          </tr>
          <tr
            class="infoRow spacingRow"
          >
            <td
              class="infoLabel"
            >
              Tapahtumien ilmoitukset:
            </td>
            <td>
              Tekstiviestillä, sähköpostilla, kieli: suomi
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
